### PR TITLE
Add disable path overlap validation flag

### PR DESCRIPTION
**Pull request by Fsero**
## What this PR does / why we need it:

In https://github.com/kubernetes/ingress-nginx/issues/5651 there was a request to throw an error when there are two ingresses defining the same host and path, which was implemented as part of the validation webhook.

  Despite of this there are clear rules on the ingress controller that
describes what the controller would do in [such situation (the oldest rule wins)](https://github.com/kubernetes/ingress-nginx/blob/main/docs/how-it-works.md?plain=1#L27)

  Some users are relying on this validation behaviour to prevent
misconfigurations, but there are use cases where allowing it, maybe temporarily, is helpful. Those use cases includes:

  - Splitting large ingresses objects in smaller ones https://github.com/kubernetes/ingress-nginx/issues/10820
  - Moving ingress objects between namespaces without downtime (like when you transfer an ingress from team A that owns namespace A to team B that owns namespace B) https://github.com/kubernetes/ingress-nginx/issues/10090

### CODE REVIEW

**Review by carlosrejano — APPROVED** [review]

File: internal/ingress/controller/controller.go [hunk 1]
```diff
@@ -1711,6 +1712,10 @@ func checkOverlap(ing *networking.Ingress, ingresses []*ingress.Ingress, servers
 				}
 			}
 
+			if disablePathOverlapValidation {
+				return nil
+			}
+
```
Comment (miguelbernadi) [inline internal/ingress/controller/controller.go]: Wouldn't we get the same behaviour if this check was done over the entire function call? I have not tried it, but reading the code seems we would get the same behaviour and a nicer API (no boolean parameter). Have you tested it?

File: internal/ingress/controller/controller_test.go [hunk 2]
```diff
@@ -339,6 +339,83 @@ func TestCheckIngress(t *testing.T) {
...
+			nginx.cfg.DisablePathOverlapValidation = false
+			nginx.command = testNginxTestCommand{
+				t:        t,
+				err:      nil,
+				expected: "_,example.com",
+			}
+
+			err = nginx.CheckIngress(duplicatedIngress)
+			if err == nil {
+				t.Errorf("expected errors but noone occurred")
```
Comment (miguelbernadi) [inline internal/ingress/controller/controller_test.go]:
```suggestion
				t.Errorf("expected errors but none occurred")
```

File: internal/ingress/controller/controller_test.go [hunk 3]
```diff
@@ -339,6 +339,83 @@ func TestCheckIngress(t *testing.T) {
...
+			}
+			t.Run("if disablePathOverlap is enabled should not throw any error", func(t *testing.T) {
```
Comment (miguelbernadi) [inline internal/ingress/controller/controller_test.go]:
```suggestion
			t.Run("when disablePathOverlap is enabled should not throw any error", func(t *testing.T) {
```

**Review by miguelbernadi — COMMENTED** [review]
Left some nits and a suggestion that needs to be tested.

**Comment by Fsero** [comment]
@miguelbernadi addressed can you check?

**Review by miguelbernadi — APPROVED** [review]
Thank you for addressing the comments! Looks good.

**Pull request merged**